libs/frontend/util-component/src/lib/compute-component-flattened-styles.ts

```diff
@@ -51,10 +54,19 @@ export const computeComponentFlattenedStyles = (
     const sourceRawStyle = breakpointStyles[breakpointId]?.[currentPseudoClass] ?? {}
 
     Object.entries(sourceRawStyle).forEach(([css, source]) => {
-      result[css as Css] = { ...source }
-      if (includePseudoClass) {
-        ;(result[css as Css] as IRawStyleWithSource).sourcePseudoClass =
-          currentPseudoClass
+      const key = css as Css
+      // Only update the result when the source type is component custom style
+      // since custom style has higher priority, or when the existing entry is
+      // not from component custom style.
+      if (
+        source.sourceType === StyleSourceType.Custom ||
+        result[key]?.sourceType !== StyleSourceType.Custom
+      ) {
```

Comment:
This seems to fix the issue. Custom styles always have the highest precedence than any other styles while flattening the styles.